src/gar.lua

```diff
@@ -361,18 +365,40 @@ function gar.decreaseDelegateStake(gatewayAddress, delegator, qty, currentTimest
 		error("Remaining delegated stake must be greater than the minimum delegated stake amount.")
 	end
 
-	-- Withdraw the delegate's stake
+	-- Instant withdrawal logic with penalty
+	if instantWithdraw then
+		-- Unlock the tokens from the gateway and delegate
+		gateway.delegates[delegator].delegatedStake = gateway.delegates[delegator].delegatedStake - qty
+		gateway.totalDelegatedStake = gateway.totalDelegatedStake - qty
 
-	local newDelegateVault = {
-		balance = qty,
-		startTimestamp = currentTimestamp,
-		endTimestamp = currentTimestamp + gar.getSettings().delegates.withdrawLengthMs,
-	}
+		-- Calculate the penalty amount
+		local maxPenalty = 0.80
+		local penaltyAmount = qty * maxPenalty
+		local amountToWithdraw = qty - penaltyAmount
 
-	-- Lock the qty in a vault to be unlocked after withdrawal period and decrease the gateway's total delegated stake
-	gateway.delegates[delegator].vaults[messageId] = newDelegateVault
-	gateway.delegates[delegator].delegatedStake = gateway.delegates[delegator].delegatedStake - qty
-	gateway.totalDelegatedStake = gateway.totalDelegatedStake - qty
+		-- Add penalty to AR.IO protocol balance
+		balances.increaseBalance(ao.id, penaltyAmount)
+
+		-- Withdraw the remaining tokens to the delegate
+		balances.increaseBalance(delegator, amountToWithdraw)
```

Comment:
shouldn't this be `decreaseBalance`?

Reply:
Nope.  The user had staked tokens, so we "unlock them", add the penalty amount to the protocol balance, and the rest to the user, and then delete the vault.

Reply:
ah, right